### DISCUSSION TITLE
Building Multi-Tenent App - vercel CNAME always same?

### DISCUSSION THREAD

**Discussion by chavda-bhavik** [vercel/community#2565] [Help] [answered]
### Summary

I'm building a multi-tenant app, When I add a domain using API, the dashboard shows two options to verify the domain,
![image](https://github.com/vercel/community/assets/50201755/2bce19a1-51bf-437e-9904-9c4b07d84504)
1. CNAME
2. Nameservers

I have doubts that, 
1. Whether CNAME will be always the same? Can I hardcode in my app that whenever a user adds a domain they can add a CNAME record to verify their domain?
2. When TXT records are needed?

![image](https://github.com/vercel/community/assets/50201755/f01c7724-a3d6-41e3-a7ed-5f4b52356347)

Thank you for your help.

**Comment by amyegan** [comment, marked as answer]
Hi @chavda-bhavik! You can generally expect that a new domain will need this CNAME record. When using the API to [add a domain to a project](https://vercel.com/docs/rest-api/endpoints#add-a-domain-to-a-project), the endpoint will return a list of verification challenges to let you know if a TXT record is needed to verify the domain. It is needed to verify ownership of a domain if, as shown in the second screenshot, another Vercel account is already using the domain.

**Reply by chavda-bhavik** [reply]
Thanks @amyegan for the explanation. 

Based on my observation of the `Add domain` API, It follows two cases

1. When any other account is not using the domain, it returns the following response. With `verified: true`.
![image](https://github.com/vercel/community/assets/50201755/d5a9436b-a300-4e8b-83f9-837b73d7e8f5)
**It means we need to add a `CNAME` record.**

2. When a domain is being used, API returns the following response with `verified: false`.
![image](https://github.com/vercel/community/assets/50201755/464abc09-197c-4ac7-a25e-8925740657f2)
**It means we need to add a `TXT` record.**

Am I going in the right direction?

**Reply by amyegan** [reply]
That's right 😄

**Comment by chavda-bhavik** [comment]
Hey @amyegan Can you help me with some documentation around when which configuration is needed? Like CNAME, TXT, A Record, etc.

Because here in this response you can see that API returns different `configurations` and `dashboard` shows different `configurations`.
# API Response
![image](https://github.com/vercel/community/assets/50201755/d8e4a135-a2e9-420d-8485-933abf4fcc0a)

# Vercel Dashboard output
![image](https://github.com/vercel/community/assets/50201755/b3e4cd93-a595-483d-92e1-433117068661)

**Reply by smaeda-ks** [reply]
@chavda-bhavik what you see in the API response is your current DNS records set to the domain.

```
$ dig CNAME www.ecomaman.com +noall +ans
www.ecomaman.com.	14400	IN	CNAME	ecomaman.com.
```

This is why you get the error on the UI because it's not the one we expect. As our UI suggests, please update your CNAME record to point to Vercel `cname.vercel-dns.com`.

And for the Apex domain, please create a `A` record with the value `76.76.21.21`.
https://vercel.com/docs/concepts/projects/domains/troubleshooting#working-with-apex-domain
